src/Microsoft.Framework.BufferedHtmlContent.Sources/BufferedHtmlContent.cs

```diff
@@ -0,0 +1,136 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.IO;
+using Microsoft.AspNet.Html.Abstractions;
+using Microsoft.Framework.WebEncoders;
+
+namespace Microsoft.Framework.Internal
+{
+    /// <summary>
+    /// Enumerable object collection which knows how to write itself.
+    /// </summary>
+    internal class BufferedHtmlContent : IHtmlContent, IEnumerable<object>
```

Comment:
The main counter-argument to not implementing `IEnumerable<T>` would have been the strangeness of a `BufferEntryCollection` that did not expose itself as a collection. Given the rename :+1: 

The `Entries` property is already `public` (called `internal` but this is an `internal` class so that's irrelevant). That means the `IEnumerable<object>` implementation is redundant. Just get rid of the `internal for testing` comment if product code e.g. the `DefaultTagHelperOutput` implementation uses `Entries`.

One option to avoid any product code using the enumeration would be `DefaultTagHelperOutput` calculating the `IsWhiteSpace` and `IsEmpty` values in the `Append()`, `AppendFormat()` and `Clear()` methods. Bit more code and that code wouldn't really be needed in the 80% case. But the code would be rather straighforward. The worst case:
``` c#
    public override TagHelperContent Append(IHtmlContent htmlContent)
    {
        var tagHelperContent = htmlContent as TagHelperContent;
        if (tagHelperContent != null)
        {
            if (!tagHelperContent.IsEmpty)
            {
                _isEmpty = false;
            }
            if (!tagHelperContent.IsWhiteSpace)
            {
                _isWhiteSpace = false;
            }
        }
        else
        {
            var value = htmlContent as string;
            if (value == null)
            {
                // Mentioned GetRawString() in Razor PR.
                value = htmlContent.GetRawString();
            }

            if (string.IsNullOrEmpty(value))
            {
                // no-op
            }
            else if (string.IsNullOrWhiteSpace(value))
            {
                _isEmpty = false;
            }
            else
            {
                _isEmpty = false;
                _isWhiteSpace = false;
            }
        }

        Buffer.Append(htmlContent);
        return this;
    }
```